apps/cowswap-frontend/src/modules/limitOrders/updaters/QuoteObserverUpdater/index.tsx

```diff
@@ -23,10 +27,21 @@ export function QuoteObserverUpdater() {
 
   const { price, isLoading } = useSpotPrice(inputToken, outputToken)
 
-  useEffect(() => {
+  // Update market rate based on spot prices
+  useSafeEffect(() => {
     updateLimitRateState({ marketRate: price, isLoadingMarketRate: isLoading })
   }, [price, isLoading, updateLimitRateState])
 
+  const { response } = useTradeQuote()
+  const { quote } = response || {}
+  const { feeAmount: feeAmountRaw } = quote || {}
+  const feeAmount = inputCurrency && feeAmountRaw ? CurrencyAmount.fromRawAmount(inputCurrency, feeAmountRaw) : null
+
+  // Update fee amount based on quote response
+  useSafeEffect(() => {
+    updateLimitRateState({ feeAmount })
+  }, [feeAmount, updateLimitRateState])
```

Comment:
We need to have the fee stored in the state for the estimated execution price calculation

Note: this fee is volume aware! In the future or as follow up, we could quote a fixed USD amount for partial fills.

Reply:
>this fee is volume aware
I think it worth it to reflect this fact in the value name.
`feeAmount` -> `feeAmountFromQuote` or smth else